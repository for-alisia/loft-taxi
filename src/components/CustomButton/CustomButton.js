import React from 'react';
import PropTypes from 'prop-types';
import './CustomButton.scss';

const CustomButton = ({ children, type, classes, handleClick }) => {
    const className = ['form-submit'];
    if (classes) {
        className.push(classes);
    }
    return (
        <button
            className={className.join(' ')}
            type={type}
            onClick={handleClick}
        >
            {children}
        </button>
    );
};

CustomButton.propTypes = {
    type: PropTypes.string,
    classes: PropTypes.string,
    handleClick: PropTypes.func
};

export default CustomButton;
