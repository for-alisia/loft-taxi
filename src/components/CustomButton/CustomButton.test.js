import React from 'react';
import CustomButton from './CustomButton';
import { shallow } from 'enzyme';

describe('Custom Button component', () => {
    const commonClass = '.form-submit';
    it('renders custom button', () => {
        expect(shallow(<CustomButton />)).toMatchSnapshot();
    });

    it('adds correct classes to button', () => {
        const wrapper = shallow(<CustomButton classes='inverse-btn' />);
        expect(wrapper.find(commonClass).hasClass('inverse-btn')).toEqual(true);
    });

    it('has correct type of button', () => {
        const wrapper = shallow(<CustomButton type='submit' />);
        expect(wrapper.find('[type="submit"]')).toHaveLength(1);
    });

    it('calls click function', () => {
        const mockHandleClick = jest.fn();
        const wrapper = shallow(<CustomButton handleClick={mockHandleClick} />);
        wrapper.find(commonClass).simulate('click');
        expect(mockHandleClick.mock.calls.length).toBe(1);
    });

    it('renders correct label', () => {
        const wrapper = shallow(<CustomButton>Label</CustomButton>);
        expect(wrapper.find(commonClass).text()).toEqual('Label');
    });
});
