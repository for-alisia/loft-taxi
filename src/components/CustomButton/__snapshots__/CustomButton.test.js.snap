// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Button component renders custom button 1`] = `ShallowWrapper {}`;
